BUG: Numpy Typing does not recognize a deque as a valid input for np.max

### Describe the issue:

When running mypy on the example code one gets the following message:

error: No overload variant matches argument type "deque[int]"  [call-overload]

Nonetheless, the code executes correctly. Should collections' deques be a valid input for both np.min and np.max?

When trying to convert a deque to a numpy array the conversions is successful; therefore, shouldn't deques be considered as array_like?

### Reproduce the code example:

```python
import numpy as np
from collections import deque

a = deque([0])
np.max(a)
```


### Error message:

```shell
error: No overload variant matches argument type "deque[int]"  [call-overload]
```


### Runtime information:

1.23.2
3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]


### Context for the issue:

_No response_

--- a/numpy/_typing/_nested_sequence.py
+++ b/numpy/_typing/_nested_sequence.py
@@ -5,7 +5,6 @@ from __future__ import annotations
 from collections.abc import Iterator
 from typing import (
     Any,
-    overload,
     TypeVar,
     Protocol,
     runtime_checkable,
@@ -62,12 +61,7 @@ class _NestedSequence(Protocol[_T_co]):
         """Implement ``len(self)``."""
         raise NotImplementedError
 
-    @overload
-    def __getitem__(self, index: int, /) -> _T_co | _NestedSequence[_T_co]: ...
-    @overload
-    def __getitem__(self, index: slice, /) -> _NestedSequence[_T_co]: ...
-
-    def __getitem__(self, index, /):
+    def __getitem__(self, index: int, /) -> _T_co | _NestedSequence[_T_co]:
         """Implement ``self[x]``."""
         raise NotImplementedError
 

--- a/numpy/typing/tests/data/reveal/array_constructors.pyi
+++ b/numpy/typing/tests/data/reveal/array_constructors.pyi
@@ -1,5 +1,6 @@
 from typing import Any, TypeVar
 from pathlib import Path
+from collections import deque
 
 import numpy as np
 import numpy.typing as npt
@@ -26,6 +27,7 @@ reveal_type(np.array(A))  # E: ndarray[Any, dtype[{float64}]]
 reveal_type(np.array(B))  # E: ndarray[Any, dtype[{float64}]]
 reveal_type(np.array(B, subok=True))  # E: SubClass[{float64}]
 reveal_type(np.array([1, 1.0]))  # E: ndarray[Any, dtype[Any]]
+reveal_type(np.array(deque([1, 2, 3])))  # E: ndarray[Any, dtype[Any]]
 reveal_type(np.array(A, dtype=np.int64))  # E: ndarray[Any, dtype[{int64}]]
 reveal_type(np.array(A, dtype='c16'))  # E: ndarray[Any, dtype[Any]]
 reveal_type(np.array(A, like=A))  # E: ndarray[Any, dtype[{float64}]]

